libraries/rush-lib/src/api/RushConfiguration.ts

```diff
@@ -622,6 +673,15 @@ export class RushConfiguration {
 
     this.ensureConsistentVersions = !!rushConfigurationJson.ensureConsistentVersions;
 
+    // Check if we have a subspace configuration file
+    const subspaceConfigLocation: string = path.join(this.rushJsonFolder, 'subspaces.json');
+    if (FileSystem.exists(subspaceConfigLocation)) {
+      // Try getting a subspace configuration
+      this.subspaceConfiguration = SubspaceConfiguration.loadFromConfigurationFile(subspaceConfigLocation);
+    }
```

Comment:
Can you rename `loadFromConfigurationFile` to `tryLoadFromConfigurationFile` and just return undefined if the file doesn't exist?

Reply:
And ideally this load should be async.

Reply:
Changed to tryLoadFromConfigurationFile

Reply:
Can this be made async?

Reply:
(If the other similar Rush functions are not already `async`, then we shouldn't insist that Will fixes it in this PR.)

Reply:
There is a `FileSystem.existsAsync`, and there is a `JsonFile.loadAndValidateAsync`. Anything that touches the filesystem (including JSON file loading) should ideally be async to allow those calls to be parallelized.